pkg/util/pravegacluster.go

```diff
@@ -79,3 +81,17 @@ func LabelsForPravegaCluster(pravegaCluster *v1alpha1.PravegaCluster, component
 func PravegaControllerServiceURL(pravegaCluster v1alpha1.PravegaCluster) string {
 	return fmt.Sprintf("tcp://%v.%v:%v", ServiceNameForController(pravegaCluster.Name), pravegaCluster.Namespace, "9090")
 }
+
+func PvcIsOrphan(stsPvcName string, replicas int32) bool {
+	index := strings.LastIndexAny(stsPvcName, "-")
```

Comment:
Can the pvc name contain another occurrence of `-` that could confuse this logic?

Reply:
Good point. If a pvc is started by a statefulset, it will have a name in this format [pvcName]-[statfulsetName]-[ordinal]. So I think there won't be any more - occur.

Reply:
Correct. StatefulSet PVC names always end with `-` + an ordinal number by design. They can contain more `-`, but the rightmost `-` will always be the separator, and that's what we capture in the `strings.LastIndexAny` function.